---
title: "lncRNA Description"
author: "Yaamini Venkataraman"
date: "5/21/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo()
```

# Generate lncRNA track

```{bash}
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/Crassostrea_virginica/GFF/ref_C_virginica-3.0_top_level.gff3.gz > ref_C_virginica-3.0_top_level.gff3.gz #Download top-level file
gunzip ref_C_virginica-3.0_top_level.gff3.gz
head ref_C_virginica-3.0_top_level.gff3
```

```{bash}
grep "gbkey=ncRNA" ref_C_virginica-3.0_top_level.gff3 > C_virginica-3.0_Gnomon_lncRNA_yrv.gff3 #Parse out lncRNA and associated exons.
```

```{bash}
head C_virginica-3.0_Gnomon_lncRNA_yrv.gff3
```

```{bash}
wc -l C_virginica-3.0_Gnomon_lncRNA_yrv.gff3 #Number of lncRNA in the C.v genome
```

# Describe lncRNA

```{bash}
awk '{print $1","$3","$4","$5}' C_virginica-3.0_Gnomon_lncRNA_yrv.gff3 > C_virginica-3.0_Gnomon_lncRNA_yrv.csv #Generate a .csv file
```

```{bash}
head C_virginica-3.0_Gnomon_lncRNA_yrv.csv
```

```{r}
lncRNA <- read.csv("C_virginica-3.0_Gnomon_lncRNA_yrv.csv", header = FALSE) #Import lncRNA csv file
colnames(lncRNA) <- c("")
head(lncRNA)
```

